### PR TITLE
Health API: Refactor the retrieval of the preflight indicators

### PR DESCRIPTION
**Pull request by gmarouli**
In this PR we refactor the definition of preflight health indicators.

**Previous implementation**
The `HealthService` constructor receives two lists of health indicators, one contains the preflight indicators and one contains the rest.

In `Node.java`, we create the lists when we create the `HealthService`. Consequently, all preflight indicators need to be "known" to `Node.java`.

**Proposed implementation**
The `HealthIndicatorService` has a `isPreflight` property. The `HealthService` receives a list of all the health indicators and then it uses this property to split them into two lists, the preflight one and the rest. This allows us to define if a health indicator is a preflight one at the `HealthIndicatorService` level.

### CODE REVIEW

**Comment by gmarouli** [comment]
@elasticmachine run elasticsearch-ci/bwc

File: server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java [hunk 1]
```diff
@@ -55,4 +55,14 @@ default HealthIndicatorResult createIndicator(
             .collect(Collectors.toList());
         return new HealthIndicatorResult(name(), status, symptom, details, impactsList, diagnosisList);
     }
+
+    /**
+     * A preflight indicator is an indicator that is run first and represents a serious cascading health problem. For example, the
+     * `stable_master` health indicator is a preflight indicator. When it is red it means that we do not have a formed cluster so running
+     * further diagnostics is not valuable.
```
Comment (andreidan) [inline server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java:62]: I believe this is not entirely correct - a node has witnessed too many maser nodes which could mean there are missing nodes, or a discovery problem, or that the node itself has problems joining the elected master. 
Furthermore, the reason all the other indicators turn to `unknown` is that the local cluster state is likely not-up-to date enough for us to make health decisions off of it.

**Review by andreidan — APPROVED** [review]
Thanks Mary, left one javadoc suggestion. 🚀

File: server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java (hunk 1 above)
Comment (gmarouli) [reply server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java:62]: Thank you @andreidan, very good point I will adjust it.

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
@elasticmachine update branch

**Comment by gmarouli** [comment]
I see the same failures in CI of other branches.

**Pull request merged**